p4-fusion/git_api.cc

```diff
@@ -286,17 +274,16 @@ std::string GitAPI::WriteChangelistBranch(
 			git_index_entry entry = {
 				.mode = GIT_FILEMODE_BLOB,
 				.path = file.GetRelativePath().c_str(),
-				// .file_size = contents.size(),
 			};
+
+			auto& blobOID = file.GetBlobOID();
+			GIT2(git_oid_fromstr(&entry.id, blobOID.c_str()));
+
 			if (file.IsExecutable())
 			{
 				entry.mode = GIT_FILEMODE_BLOB_EXECUTABLE; // 0100755
 			}
 
-			git_oid objId;
-			GIT2(git_blob_create_from_buffer(&objId, m_Repo, contents.data(), contents.size()));
-			GIT2(git_oid_cpy(&entry.id, &objId));
-
```

Comment:
this is the goodie. no more keeping the contents in memory until the CL is being processed by the main thread. Saves gigabytes of memory on a small-ish depot (9000CL, no large files)